### PR TITLE
[bitnami/postgresql] Postgresql TLS Support

### LINKED ISSUE #2812: [bitnami/postgresql] How to enable ssl and external access through ingress
Hi Team,

I was enabling ssl/tls on postgresql but pod is failing to come up with failure related to password authentication.

I have copied certificates to postgresql data directory and added ssl=on in postgresql.conf

Following is the error on pod startup.
```
postgresql 08:46:05.95 INFO  ==> Loading user's custom files from /docker-entrypoint-initdb.d ...
postgresql 08:46:05.96 INFO  ==> Starting PostgreSQL in background...
waiting for server to start....2020-06-11 08:46:05.986 GMT [136] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2020-06-11 08:46:05.986 GMT [136] LOG:  could not bind IPv6 address "::1": Cannot assign requested address
2020-06-11 08:46:05.986 GMT [136] HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
2020-06-11 08:46:05.990 GMT [136] LOG:  listening on Unix socket "/tmp/.s.PGSQL.5432"
2020-06-11 08:46:06.002 GMT [137] LOG:  database system was shut down at 2020-06-11 08:46:03 GMT
2020-06-11 08:46:06.007 GMT [136] LOG:  database system is ready to accept connections
 done
server started
2020-06-11 08:46:06.076 GMT [145] FATAL:  password authentication failed for user "postgres"
2020-06-11 08:46:06.076 GMT [145] DETAIL:  User "postgres" has no password assigned.
	Connection matched pg_hba.conf line 3: "local    all             all                                     md5"
2020-06-11 08:46:06.076 GMT [145] LOG:  could not send data to client: Broken pipe
postgresql 08:46:06.08 DEBUG ==> Executing /docker-entrypoint-initdb.d/security_ddl.sql
2020-06-11 08:46:06.092 GMT [151] FATAL:  password authentication failed for user "postgres"
2020-06-11 08:46:06.092 GMT [151] DETAIL:  User "postgres" has no password assigned.
	Connection matched pg_hba.conf line 1: "host     all             all             0.0.0.0/0               md5"
2020-06-11 08:46:06.094 GMT [152] FATAL:  password authentication failed for user "postgres"
2020-06-11 08:46:06.094 GMT [152] DETAIL:  User "postgres" has no password assigned.
	Connection matched pg_hba.conf line 1: "host     all             all             0.0.0.0/0               md5"
psql: FATAL:  password authentication failed for user "postgres"
FATAL:  password authentication failed for user "postgres"
postgresql 08:46:06.09 INFO  ==> Stopping PostgreSQL...
2020-06-11 08:46:06.100 GMT [136] LOG:  received smart shutdown request
2020-06-11 08:46:06.104 GMT [136] LOG:  background worker "logical replication launcher" (PID 143) exited with exit code 1
2020-06-11 08:46:06.104 GMT [138] LOG:  shutting down
2020-06-11 08:46:06.123 GMT [136] LOG:  database system is shut down
```
Thanks in advance!

### PR DESCRIPTION
**Pull request by joancafom**
**Description of the change**

**Benefits**
Adds the possibility to secure traffic using TLS.

**Possible drawbacks**
- None, regular traffic is still supported.

**Applicable issues**

  - fixes #2812

**Additional information**
An init-container was needed to ensure certificate-related files have the right permissions. Mounting the secret using `defaultMode` or `mode` options did end up into PostgreSQL having issues to start. This is due to an existing [issue](https://github.com/kubernetes/kubernetes/issues/57923) in K8s that makes those options conflict when `securityContext.runAsUser` is used. A warning in README.md was added explaining this.

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/postgresql/README.md [hunk 1]
```diff
@@ -329,6 +336,35 @@ In addition to these options, you can also set an external ConfigMap with all th
...
+
+For example:
+
+First, create the secret with the cetificates files:
+
+```console
+kubectl create secret generic certificates-tls-secret --from-file=./cert.crt --from-file=./cert.key --from-file=./ca.crt
+```
+
+Then, use the following parameters:
```
Comment (marcosbc) [inline bitnami/postgresql/README.md:356]: We could use a list for this (and intent the code) to make it more readable, i.e.:

Refer to the following example:

* First, create the secret with the certificate files:

    ```console
    kubectl create secret ...
    ```

* Then, install the chart with the following parameters:

    ```console
    volumePermissions.enabled=true
    ...

    ```

    > NOTE: PostgreSQL requires certain permissions on sensitive files (such as certificate keys) to start up. Due to an on-going [issue](https://github.com/kubernetes/kubernetes/issues/57923) regarding kubernetes permissions and the use of `securityContext.runAsUser`, you must enable `volumePermissions` to ensure everything works as expected.

___

```
Refer to the following example:

* First, create the secret with the certificate files:

    ```console
    kubectl create secret ...
    ```

* Then, install the chart with the following parameters:

    ```console
    volumePermissions.enabled=true
    ...

    ```

    > NOTE: PostgreSQL requires certain permissions on sensitive files (such as certificate keys) to start up. Due to an on-going [issue](https://github.com/kubernetes/kubernetes/issues/57923) regarding kubernetes permissions and the use of `securityContext.runAsUser`, you must enable `volumePermissions` to ensure everything works as expected.
```

File: bitnami/postgresql/files/certs/docker-entrypoint-initdb.d/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+You can copy here your custom `.crt`, `.key` or `.crl` files to easily create a secret out of them.
```
Comment (marcosbc) [inline bitnami/postgresql/files/certs/docker-entrypoint-initdb.d/README.md:1]: AFAIK we decided not to include these initdb.d folders anymore, as they were confusing to users and required the chart source or repo to be installed on the file system.

File: bitnami/postgresql/templates/statefulset-slaves.yaml [hunk 3]
```diff
@@ -95,6 +95,15 @@ spec:
               {{- if and .Values.shmVolume.enabled .Values.shmVolume.chmod.enabled }}
               chmod -R 777 /dev/shm
               {{- end }}
+              {{- if .Values.tls.enabled }}
+              cp /tmp/certs/* /opt/bitnami/postgresql/certs/
```
Comment (marcosbc) [inline bitnami/postgresql/templates/statefulset-slaves.yaml:99]: Is it guaranteed that there will be a file inside that folder? Note that dotfiles (`.file`) and files inside a subdirectory will not be copied with this `cp` command.

File: bitnami/postgresql/templates/statefulset-slaves.yaml [hunk 4]
```diff
@@ -95,6 +95,15 @@ spec:
...
+              {{- if eq ( toString ( .Values.volumePermissions.securityContext.runAsUser )) "auto" }}
+                chown -R `id -u`:`id -G | cut -d " " -f2` /opt/bitnami/postgresql/certs/
+              {{- else }}
+                chown -R {{ .Values.securityContext.runAsUser }}:{{ .Values.securityContext.fsGroup }} /opt/bitnami/postgresql/certs/
```
Comment (marcosbc) [inline bitnami/postgresql/templates/statefulset-slaves.yaml:103]: I would not indent the code block here, as it is in the same level than the `cp` command.

File: bitnami/postgresql/README.md (hunk 1 above)
Comment (joancafom) [reply bitnami/postgresql/README.md:356]: Okay, sounds nice to me!

File: bitnami/postgresql/files/certs/docker-entrypoint-initdb.d/README.md (hunk 2 above)
Comment (joancafom) [reply bitnami/postgresql/files/certs/docker-entrypoint-initdb.d/README.md:1]: 👍 had no idea, removing this option

File: bitnami/postgresql/templates/statefulset-slaves.yaml (hunk 3 above)
Comment (joancafom) [reply bitnami/postgresql/templates/statefulset-slaves.yaml:99]: Yes, when `.Values.tls.enabled=true` we have some templates in `_helpers.tpl` than enforce the existence of cert files. 

E.g.:

```yaml
{{- define "postgresql.tlsCert" -}}
{{- required "Certificate filename is required when TLS in enabled" .Values.tls.certFilename | printf "/opt/bitnami/postgresql/certs/%s" -}}
{{- end -}}
```

Those cert files are mounted in `/tmp/certs/` in the init-container. Regarding dotfiles, the main goal of this init-container workaround is to set the right permissions to `.crt`, `.key` and `.crl` files, so no issues should be expected with respect to that.

File: bitnami/postgresql/templates/statefulset-slaves.yaml (hunk 4 above)
Comment (joancafom) [reply bitnami/postgresql/templates/statefulset-slaves.yaml:103]: Got it! Thanks

**Pull request merged**